addons/website_event_booth_exhibitor/controllers/event_booth.py

```diff
@@ -30,37 +29,13 @@ def _prepare_booth_registration_partner_values(self, event, kwargs):
         return super(WebsiteEventBoothController, self)._prepare_booth_registration_partner_values(event, kwargs)
 
     def _prepare_booth_registration_sponsor_values(self, event, booth_values, kwargs):
-        booth_category = request.env['event.booth.category'].sudo().search(
-            [('id', '=', int(kwargs['booth_category_id']))]
-        )
-        if booth_category.use_sponsor:
-            sponsor = request.env['event.sponsor'].sudo().search([
-                ('partner_id', '=', booth_values.get('partner_id')),
-                ('sponsor_type_id', '=', booth_category.sponsor_type_id.id),
-                ('exhibitor_type', '=', booth_category.exhibitor_type),
-                ('event_id', '=', event.id),
-            ], limit=1)
-            if not sponsor:
-                sponsor_name = kwargs.get('sponsor_name') or booth_values.get('contact_name')
-                sponsor_values = {
-                    'email': kwargs.get('sponsor_email') or booth_values.get('contact_email'),
-                    'event_id': event.id,
-                    'exhibitor_type': booth_category.exhibitor_type,
-                    'name': sponsor_name,
-                    'partner_id': booth_values['partner_id'],
-                    'mobile': kwargs.get('sponsor_mobile') or booth_values.get('contact_mobile'),
-                    'phone': kwargs.get('sponsor_phone') or booth_values.get('contact_phone'),
-                    'sponsor_type_id': booth_category.sponsor_type_id.id,
-                    'subtitle': kwargs.get('sponsor_slogan'),
-                    'website_description': plaintext2html(kwargs.get('sponsor_description')),
-                    'image_512': base64.b64encode(kwargs['sponsor_image'].read()) if kwargs.get('sponsor_image') else False,
-                }
-                if booth_category.exhibitor_type == 'online':
-                    sponsor_values.update({
-                        'room_name': 'odoo-exhibitor-%s' % sponsor_name,
-                    })
-                sponsor = request.env['event.sponsor'].sudo().create(sponsor_values)
-            return {
-                'sponsor_id': sponsor.id,
-            }
-        return {}
+        sponsor_values = {
+            'sponsor_name': kwargs.get('sponsor_name') or booth_values.get('contact_name'),
+            'sponsor_email': kwargs.get('sponsor_email') or booth_values.get('contact_email'),
+            'sponsor_mobile': kwargs.get('sponsor_mobile') or booth_values.get('contact_mobile'),
+            'sponsor_phone': kwargs.get('sponsor_phone') or booth_values.get('contact_phone'),
+            'sponsor_subtitle': kwargs.get('sponsor_slogan'),
+            'sponsor_website_description': plaintext2html(kwargs.get('sponsor_description')),
```

Comment:
Just to be sure, is everything ok if sponsor_description is void / False / ... ? Looking at code, not sure. Maybe we could be sure and just say ``'sponsor_website_description': plaintext2html(kwargs['sponsor_description']) if kwargs.get('sponsor_description') else ''`` ?